app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,108 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList> structRepo = new HashMap<>();
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        process(translationUnit);
+    }
+
+    private void process(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child : children) {
+            if (child instanceof CPPASTSimpleDeclaration) {
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+
+                if (declaration.getDeclSpecifier() instanceof CPPASTCompositeTypeSpecifier) {
+                    CPPASTCompositeTypeSpecifier compSpec = (CPPASTCompositeTypeSpecifier) declaration.getDeclSpecifier();
+                    CPPASTName name = (CPPASTName) compSpec.getName();
+                    saveStuct(compSpec.getDeclarations(false), name.getRawSignature());
+                } else {
+                    saveConst(child);
+                }
+
+            }
+        }
+    }
+
+    private ArrayList getComponents(IASTNode node) throws Exception {
+        ArrayList components = new ArrayList();
+        CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) node;
+        CPPASTNamedTypeSpecifier specifier = (CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier();
+        CPPASTDeclarator declarator = (CPPASTDeclarator) declaration.getDeclarators()[0];
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+
+        if (declarator != null || specifier != null || declaration != null && init != null) {
```

Comment:
Is this the correct logic? Under what cases should things be allowed to be null?